### ISSUE TITLE
Can't delete NotUploaded templates (upload from local)

### ISSUE THREAD

**Issue by andrijapanicsb** [apache/cloudstack#8375]
![image](https://github.com/apache/cloudstack/assets/45762285/d6ff06ba-5dd1-432a-b532-cdb40e62fc37)

Reproduce: Uploading template from local (no SSL config, it's using default realhostip.com SSL) - since SSL is untrusted, the browser can not connect - but template seems created in NotUploaded state - and can't be deleted.

**Comment by weizhouapache** [comment]
@andrijapanicsb 

wait a hour :-D
when the state is changed to 'UploadError', you will be able to delete the template

**Comment by andrijapanicsb** [comment]
Copy that, good workaround :)
Not sure why we even created DB record, would be nice to improve this.

**Comment by DaanHoogland** [comment]
seems like `state==NotUploaded` is a valid starting point for removal. I think we can add that to the template lifecycle.

**Comment by weizhouapache** [comment]
> NotUploaded

@DaanHoogland 
it is the initial state of template. The final state could be UploadAbandoned, Active, UploadError
https://github.com/apache/cloudstack/blob/main/api/src/main/java/com/cloud/template/VirtualMachineTemplate.java#L50C113-L56

**Comment by DaanHoogland** [comment]
> @DaanHoogland it is the initial state of template. The final state could be UploadAbandoned, Active, UploadError

meaning we cannot safely assume we can abandon and delete it?

**Comment by weizhouapache** [comment]
> > @DaanHoogland it is the initial state of template. The final state could be UploadAbandoned, Active, UploadError
> 
> meaning we cannot safely assume we can abandon and delete it?

right. the state is changed to UploadAbandoned, only if it times out after 1 hour  (or global setting upload.operation.timeout)
there is no api to abandon it and update the state to UploadAbandoned
